python/ray/serve/autoscaling_metrics.py

```diff
@@ -1,17 +1,21 @@
 import threading
 import bisect
 from collections import defaultdict
+import logging
 import time
 from typing import Callable, DefaultDict, Dict, List, Optional
 from dataclasses import dataclass, field
 
 import ray
 
+logger = logging.getLogger(__file__)
+
 
 def start_metrics_pusher(
     interval_s: float,
     collection_callback: Callable[[], Dict[str, float]],
-    controller_handle,
+    metrics_process_func,
+    stop_event=None,
 ):
     """Start a background thread to push metrics to controller.
```

Comment:
Can you add some info about what conditions cause this thread to stop, now that we've introduce a `stop_event`?